### ISSUE TITLE
Could you add a parameter to enable 90-degree screen rotation?

### ISSUE THREAD

**Issue by stanley700206** [jlobue10/SteamDeck_rEFInd#51]
Hi
My English is not good...

Could you add a parameter to enable 90-degree screen rotation?
https://drive.google.com/file/d/1nEVC8VzJmIOsme-pMHl1_n-J9_ce9Aif/view?usp=share_link
My handheld computer model is Ayaneo2

**Comment by jlobue10** [comment]
I'll do my best to support, but I must add the disclaimer that the AyaNeo 2 is not the desired target for this repo. Many code assumptions are done based off of the specific configuration and needs of the Steam Deck.

That 90 degree rotation most of the time comes down to the `resolution` line in the `refind.conf` config file. The setting that works really well for the Steam Deck is `resolution 3`. I don't know if this same setting would work well for the AyaNeo 2, but it'd definitely be worth trying as a potential simple fix (assuming that you are already using rEFInd).

**Comment by stanley700206** [comment]
Thank you for your response. I have tried the resolution  3, but it doesn't work on my Ayaneo2, which is why I wrote to you hoping for a way to rotate the screen by 90 degrees. I am using rEFInd, which is a fantastic software.

**Comment by jlobue10** [comment]
Okay, so that extra information helps. I'm not the original rEFInd author. This script only makes installation and configuration for Steam Deck rEFInd about as easy as possible.

Another thing that I would try is either commenting the `resolution` line out, or deleting it altogether. If this works as I'd expect, you would get an EFI pop-up screen on boot with possible resolutions. If you see this resolution screen, maybe one of those will get you the 90 degree rotation that you're after. It'll be down to trial and error for you, if this possible solution works or not.

**Comment by stanley700206** [comment]
I am very willing to cooperate with the testing, and if there are any issues with the test, I will provide you with feedback.

Thank you for your response.

[https://drive.google.com/file/d/1diwjcBhjtRQX5uwXUq-sDFIfcHw8M3Db/view?usp=sharing](url)

**Comment by jlobue10** [comment]
I would try `resolution 0`. If that doesn't fix the 90 degree rotation, then try `resolution 1`. One of those settings seems like it would be the 90 degree shift that you're after (hopefully).

**Comment by stanley700206** [comment]
Thank you for your response.
The following is the result of my tes
resolution 0
https://drive.google.com/file/d/1BJOB5b92R-20-XYdyzPxAXSgv5oAgcIk/view?usp=sharing
resolution 1
https://drive.google.com/file/d/1Hlp9SH8FyhoBjgEQvfRcRt5BJLqpLI_w/view?usp=sharing
resolution 3
https://drive.google.com/file/d/1hTYZxWbGClyN9tqKiSO1nBF9pi9zuOaI/view?usp=sharing

**Comment by jlobue10** [comment]
Yes, sorry none of those work out. The rEFInd bootloader author actually lists this as one of his [nice to have todo items](https://www.rodsbooks.com/refind/todo.html). There's not really anything I can do about it, as the fix would have to come from him.

**Comment by jlobue10** [comment]
This "fix" or ability will have to come from the original rEFInd boot loader developer. Closing the issue.

**Comment by DanaGoyette** [comment]
I seem to recall, if you set "resolution X" for X being some silly large number that probably doesn't exist, refind will stop and print a list of the known modes.
It's possible there are well more than 3 modes on your device.
